tools/optimizer/optimizer.cpp

```diff
@@ -3869,6 +3869,31 @@ void asmLastOpts(Ref ast) {
   });
 }
 
+// Contrary to the name this does not eliminate actual dead functions, only
+// those marked as such with DEAD_FUNCTIONS
+void eliminateDeadFuncs(Ref ast) {
+  assert(!!extraInfo);
+  IString DEAD_FUNCTIONS("dead_functions");
+  IString ABORT("abort");
+  assert(extraInfo->has(DEAD_FUNCTIONS));
+  Ref dead_functions = extraInfo[DEAD_FUNCTIONS];
+  traverseFunctions(ast, [&](Ref fun) {
+    for (size_t i = 0; i < dead_functions->size(); i++) {
+        if (dead_functions[i] == fun[1].get()->getCString()) {
+            break;
+        }
+        return;
+    }
+    AsmData asmData(fun);
+    fun[3]->setSize(1);
+    fun[3][0] = make1(STAT, make2(CALL, makeName(ABORT), makeArray()));
+    while (asmData.vars.size() > 0) {
```

Comment:
I guess we don't care about leaving the values in .locals because the structure gets thrown away anyway?